### PR TITLE
Don't log 4XX errors as error-level in our logs for right now

### PR DESCRIPTION
**Pull request by willh-meetup**

### CODE REVIEW

File: src/plugins/api-proxy/util/receive.js [hunk 1]
```diff
@@ -196,31 +200,32 @@ export const makeLogResponse = request => ([response, body]) => {
...
+				id,
+				statusCode,
+				time: elapsedTime,
+				originRequestId: request.id,
+				body,
+			},
+		});
+
+		// Only want to warn about 400s for now (maybe forever?)
+		if (statusCode < 500) {
```
Comment (willh-meetup) [inline src/plugins/api-proxy/util/receive.js]: Truthfully, I'm not sure if new relic picks these up, but I think better to err on the side of caution.

File: src/plugins/api-proxy/util/receive.js [hunk 2]
```diff
@@ -122,9 +122,13 @@ export const parseApiValue = ([response, body]) => {
...
 	try {
 		const value = parseBody(body);
-		if (response.statusCode < 200 || response.statusCode > 399) {
+		if (response.statusCode < 200 || response.statusCode >= 500) {
 			return formatApiError(new Error(response.statusMessage), value);
 		}
+		if (response.statusCode >= 400) {
+			// Don't throw an error for 400s for now (maybe forever?)
+			return formatApiError({ message: response.statusMessage }, value);
+		}
```
Comment (mmcgahan) [inline src/plugins/api-proxy/util/receive.js]: this has the same effect of the previous implementation - `formatApiError` expects an `Error` instance and will take the `message` prop from it without `throw`ing.

File: src/plugins/api-proxy/util/receive.js [hunk 3]
```diff
@@ -196,31 +200,32 @@ export const makeLogResponse = request => ([response, body]) => {
...
-				},
-			})
-		);
 	}
 	const logger = request.server.app.logger;
 	// production logs will automatically be JSON-parsed in Stackdriver
-	const log = ((statusCode >= 400 && logger.error) ||
-		(statusCode >= 300 && logger.warn) ||
+	const log = ((statusCode >= 500 && logger.error) ||
+		(statusCode >= 400 && logger.warn) ||
```
Comment (willh-meetup) [inline src/plugins/api-proxy/util/receive.js]: I'm betting on this *definitely* making a difference.

File: src/plugins/api-proxy/util/receive.js [hunk 4]
```diff
@@ -196,31 +200,32 @@ export const makeLogResponse = request => ([response, body]) => {
...
+				body,
+			},
+		});
+
+		// Only want to warn about 400s for now (maybe forever?)
+		if (statusCode < 500) {
+			console.warn(errorMessageString);
+		} else {
+			console.error(errorMessageString);
 		}
```
Comment (mmcgahan) [inline src/plugins/api-proxy/util/receive.js]: trivially DRYer alternative

```js
const logError = statusCode < 500 ? console.warn : console.error;
logError(errorMessageString);
```

File: src/plugins/api-proxy/util/receive.js (hunk 1 above)
Comment (mmcgahan) [reply src/plugins/api-proxy/util/receive.js]: yeah i'm guessing it watches stderr

File: src/plugins/api-proxy/util/receive.js (hunk 2 above)
Comment (willh-meetup) [reply src/plugins/api-proxy/util/receive.js]: Yeah, I did this first and it I don't think it's helpful. Removed.

File: src/plugins/api-proxy/util/receive.js (hunk 4 above)
Comment (willh-meetup) [reply src/plugins/api-proxy/util/receive.js]: 👍

**Comment by willh-meetup** [comment]
OK, comments addressed.

**Pull request merged**